pkg/controllers/subnetport/subnetport_controller.go

```diff
@@ -364,7 +366,8 @@ func deleteSuccess(r *SubnetPortReconciler, _ *context.Context, o *v1alpha1.Subn
 }
 
 func (r *SubnetPortReconciler) GetSubnetPathForSubnetPort(ctx context.Context, subnetPort *v1alpha1.SubnetPort) (string, error) {
-	subnetPath := r.SubnetPortService.GetSubnetPathForSubnetPortFromStore(string(subnetPort.UID))
+	subnetPortID := r.SubnetPortService.BuildSubnetPortId(&subnetPort.ObjectMeta)
+	subnetPath := r.SubnetPortService.GetSubnetPathForSubnetPortFromStore(subnetPortID)
```

Comment:
In [`pod_controller.go`](https://github.com/vmware-tanzu/nsx-operator/blob/6e7c54aba36a6a8a46892cd44cfb9789b0c45ded/pkg/controllers/pod/pod_controller.go#L254), there's also an invoking of `GetSubnetPathForSubnetPortFromStore`, where the `BuildSubnetPortId` should be called before it, please check it.

Reply:
updated.